File: web-pm-vertx-java/src/main/java/cn/yjl/vertx/handler/ListAccountHandler.java
package cn.yjl.vertx.handler;

import io.vertx.core.json.JsonArray;
import io.vertx.ext.web.RoutingContext;
import io.vertx.mysqlclient.MySQLPool;

/**
 * @author jiaolongyin
 * @description: TODO
 * @date 2022/12/28 11:29
 */
public class ListAccountHandler extends AbstractHandler {

    public ListAccountHandler(MySQLPool mysqlClient) {
        super(mysqlClient);
    }

    @Override
    public void handle(RoutingContext context) {
        mysqlClient
                .preparedQuery("select id, created_at, updated_at, deleted_at, act_name, act_pwd, act_nick_name, act_introduction, act_status, act_register_date from accounts")
                .execute()
                .map(rows -> {
                    JsonArray array = new JsonArray();
                    rows.forEach(row -> array.add(row.toJson()));
                    return array;
                })
                .onSuccess(array -> context.end(array.toBuffer()));
    }
}

File: web-pm-vertx/src/main/kotlin/cn/yjl/vertx/handler/ListAccountHandler.kt
package cn.yjl.vertx.handler

import cn.yjl.vertx.entity.Accounts
import cn.yjl.vertx.util.end
import io.vertx.ext.web.RoutingContext
import io.vertx.kotlin.coroutines.await
import io.vertx.sqlclient.SqlClient

class ListAccountHandler(private val mysqlClient: SqlClient): AbstractHandler {

    override suspend fun handle(context: RoutingContext) {
        val result = mysqlClient.preparedQuery("select * from accounts").execute().await()
        val entityList = result.toList().map { Accounts.fromRow(it) }
        context.end(entityList).await()
    }

}
